Base code

LexiShew opened the pull request:
Before issuing a pull request, please see the contributing page.
Essentially gets rid of everything in the team's code base except for DriveSystem, IMUSystem, motors, scales, and a sample op mode.

LexiShew committed:
Removed hardware map parameters from DriveSystem and IMUSystem. Sets the necessary components to the hardware map in op modes.

LexiShew committed:
Sample OpMode that demonstrates how to set components in the opmode (as opposed to passing in an opmode to the component) to keep components independent of the hardware map and opmode.

zerbash commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/DriveSystem.java:
```diff
@@ -6,25 +6,21 @@
-    private DcMotor motorFrontRight;
-    private DcMotor motorBackLeft;
-    private DcMotor motorBackRight;
+    public DcMotor motorFrontLeft;
```
Let's keep these private

zerbash commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/DriveSystem.java:
```diff
@@ -6,25 +6,21 @@
-    public DriveSystem(HardwareMap hardwareMap) {
-
-        this.hardwareMap = hardwareMap;
+    public DriveSystem() {
```
Let's take in an array of motors here...

zerbash commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/components/IMUSystem.java:
```diff
@@ -28,15 +28,14 @@
      * Creates a new IMU System
      */
-    public IMUSystem(HardwareMap hardwareMap)
+    public IMUSystem()
```
Here too -- let's keep this private, but set it in the constructor

zerbash commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/teleop/DriveTeleop.java:
```diff
@@ -17,7 +18,11 @@
     public void initialize()
     {
 
-        this.driveSystem = new DriveSystem(this);
```
Initialize the array of motors, then pass them in here.

LexiShew committed:
Sample OpMode that demonstrates how to set components in the opmode (as opposed to passing in an opmode to the component) to keep components independent of the hardware map and opmode.

LexiShew committed:
Some restructuring of last commit.

Last commit: Sample OpMode that demonstrates how to set components in the opmode (as opposed to passing in an opmode to the component) to keep components independent of the hardware map and opmode.
